scvi/model/_scvi.py

```diff
@@ -188,4 +200,8 @@ def setup_anndata(
             fields=anndata_fields, setup_method_args=setup_method_args
         )
         adata_manager.register_fields(adata, **kwargs)
+        # register new fields for latent mode if needed
+        latent_mode = _get_latent_adata_type(adata)
+        if latent_mode is not None:
+            adata_manager.register_new_fields(cls._get_latent_fields(latent_mode))
```

Comment:
can you check if latent mode and then append the latent fields to anndata_fields on line 185? This way validation will be performed on loading

Reply:
let's purse this change